elf/elf.cc

```diff
@@ -103,8 +103,14 @@ elf::elf(const std::shared_ptr<loader> &l)
         // Load sections
         const void *sec_data = l->load(m->hdr.shoff,
                                        m->hdr.shentsize * m->hdr.shnum);
+        if (NULL == sec_data) {
+            return;
+        }
         for (unsigned i = 0; i < m->hdr.shnum; i++) {
                 const void *sec = ((const char*)sec_data) + i * m->hdr.shentsize;
+                if (NULL == sec) {
```

Comment:
It doesn't make sense for sec to be NULL here, since it's a result from a pointer offset from sec_data. In practice the only way this can happen is if sec_data itself is NULL, but then we shouldn't have even done the pointer arithmetic.